### ISSUE TITLE
APs should be able to obtain a list of all registrations associated with a learner, activity or learner/activity pair

### ISSUE THREAD

**Issue by garemoko** [adlnet/xAPI-Spec#543]
It would be helpful for an activity provider to be able to obtain a list of all registrations that have data in the Document APIs. For example this might be used to display a list of previous attempts to the learner to choose from, or for a reporting tool reporting on state data  (with a small 's' in the sense of all the Document APIs deal with the state of play, I'm not referring explicitly to the State API). 

**_Breaking change:**_ The LRS MUST maintain a list of registrations in some JSON document we define with a key of http://adlnet.gov/xapi/keys/registration-list

**_Non-breaking change:**_ APs MAY maintain a list of registrations in some JSON document we define with a key of http://adlnet.gov/xapi/keys/registration-list

For examples of applications that might use this feature, see https://github.com/garemoko/moodle-mod_tincanlaunch and https://github.com/adlnet/xAPI-SCORM-Profile/issues/7

**Comment by garemoko** [comment]
Waiting to discuss @bscSCORM's comment on a call before PRing this.

**Comment by andyjohnson** [comment]
Discussion on 2/04/15 - Examine as a larger effort the querying purposes and capabilities of the LRS.  If we are going to enable new use cases, we should really enable them and not just solve pieces at a time (another piece brought up was the querying of Document APIs).  An exception may be timestamp sorting, which is currently impossible without pulling every Statement (can only limit by "stored").  This will be tagged as a 2.0, with timestamp sorting potentially being a 1.1.

The management use case should also be further vetted before determining if this additional requirement is helpful.

**Comment by canweriotnow** [comment]
Sounds great except for this:

> in some JSON document

Specifying the server-side representation of a resource is the AntiREST, and I think we can do better. Perhaps

* The LRS (term) maintain as a resource list of registrations we define with a key of http://adlnet.gov/xapi/keys/registration-list"
* The LRS (term) support MIME type "application/json" for representation of this resource.

Just to try to start being RESTful.

**Comment by garemoko** [comment]
Yep, I meant that it would be retrievable as json; the LRS can store the data as carefully arranged fish in a frying pan for all I care! This wasn't at all intended to be final language, though I can see how the way I laid it out might give that impression.

**Comment by canweriotnow** [comment]
@garemoko Yeah, I figured that, it's one of those editing things that I think needs to happen in a bunch of places :smile_cat:
